### PR TITLE
feat(database) building database

### PR DESCRIPTION
**Pull request by jesscxu**
set up the migrations, model, validators, and controllers

### CODE REVIEW

File: config/test.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
 module.exports = {
-  FOO: 'bar'
+  FOO: 'bar',
```
Comment (charlesbjohnson) [inline config/test.js]: Let's remove this.

File: config/test.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+  DB_HOST: 'localhost',
```
Comment (charlesbjohnson) [inline config/test.js]: nit: spacing is off here

File: db/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+const Config = require('../config');
+
+module.exports = {
+  client: 'postgresql',
+  connection: {
+  database: Config.DB_NAME,
+  host: Config.DB_HOST,
+  password: Config.DB_PASSWORD,
+  port: Config.DB_PORT,
+  user: Config.DB_USER
```
Comment (charlesbjohnson) [inline db/index.js]: nit: spacing is off here

File: db/migrations/20170405150351_create_movies.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+exports.up = function(Knex, Promise) {
```
Comment (charlesbjohnson) [inline db/migrations/20170405150351_create_movies.js]: nit: space after `function`

File: db/migrations/20170405150351_create_movies.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+exports.up = function(Knex, Promise) {
+  return Knex.schema.createTable('movies', (table) => {
+    table.increments('id').primary();
+    table.text('title').notNullable();
+    table.integer('release_year');
+  });
+};
+
+exports.down = function(Knex, Promise) {
```
Comment (charlesbjohnson) [inline db/migrations/20170405150351_create_movies.js]: nit: space after `function`

File: db/seeds/data/movies.json [hunk 6]
```diff
@@ -0,0 +1,1096 @@
...
+{
+  "title": "Big Eyes",
+  "release_year": 2014
+},
+{
+  "title": "Big Sur",
+  "release_year": 2013
+},
+{
+  "title": "Big Touble in Little China",
```
Comment (charlesbjohnson) [inline db/seeds/data/movies.json:126]: giphy

File: lib/plugins/features/movies/index.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const MovieValidator = require('../../../validators/movie');
+const Controller     = require('./controller');
```
Comment (charlesbjohnson) [inline lib/plugins/features/movies/index.js]: nit: alphabetize `require`s

**Review by charlesbjohnson — APPROVED** [review]
LGTM pending some nits

**Review by sjl2 — APPROVED** [review]
LGTM with Charlie's suggestions

**Pull request merged**